app/services/search/location_builder.rb

```diff
@@ -38,7 +35,13 @@ def location_category_search?
 
   def initialize_location_polygon
     @location_polygon = LocationPolygon.with_name(@location_category)
-    @location_polygon_boundary = [@location_polygon.boundary] if @location_polygon.present?
+    if @location_polygon.present?
+      @location_polygon_boundary = if @buffer_radius.present?
```

Comment:
I'm uncomfortable about naming the buffered polygon the 'location polygon boundary' - that sounds like it refers to the original boundary of the LocationPolygon. Would @search_polygon_boundary or @search_polygon suit your purposes, or @polygon_boundary, or @boundary, or @search_boundary...?

Reply:
I think search_polygon_boundary is probably the best option.